h/auth/services.py

```diff
@@ -61,26 +65,26 @@ def verify_ticket(self, principal, ticket_id):
         if ticket_id is None:
             return False
 
-        # Since we load and then update the AuthTicket.expires on every request,
-        # we optimise it with a `UPDATE ... RETURNING` query, doing both of
-        # these in one. When the returned `userid` is `None` then the
-        # authentication failed, otherwise it succeeded. Doing a query like this
-        # will not mark the session as dirty, and thus the transaction will
-        # get rolled back at the end, unless we found a userid, at which point
-        # we manually mark the session as changed.
-        ticket_query = (sa.update(AuthTicket.__table__)
-                        .where(sa.and_(AuthTicket.id == ticket_id,
-                                       AuthTicket.user_userid == principal,
-                                       AuthTicket.expires > sa.func.now()))
-                        .values(expires=(sa.func.now() + TICKET_TTL))
-                        .returning(AuthTicket.user_userid))
-        self._userid = self.session.execute(ticket_query).scalar()
-
-        if self._userid:
-            mark_changed(self.session)
-            return True
-
-        return False
+        ticket = self.session.query(AuthTicket) \
+            .filter(AuthTicket.id == ticket_id,
+                    AuthTicket.user_userid == principal,
+                    AuthTicket.expires > sa.func.now()) \
+            .first()
```

Comment:
Given the PK constraint on `AuthTicket.id` it might be clearer to do `.one_or_none()` here.